File: 5 - JAVA e PHP (POO)/PHP/Aula15_Exercicio_Parte_02/Visualizacao.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of Visualizacao
 *
 * @author ronan
 */

require_once 'Gafanhoto.php';
require_once 'Pessoa.php';

class Visualizacao {
    //Atributos
    private $espectador; //instância Gafanhoto
    private $filme;      //instância Video   
    
    //Construtor
    public function __construct($gafanhoto,$video){
        $this->setFilme($video);
        $this->setEspectador($gafanhoto);
        $this->espectador->viuMaisUm();
        $this->filme->setViews($this->filme->getViews() + 1);
    }
    
    //Getters and Setters 
    function getEspectador() {
        return $this->espectador;
    }

    function getFilme() {
        return $this->filme;
    }

    function setEspectador($espectador) {
        $this->espectador = $espectador;
    }

    function setFilme($filme) {
        $this->filme = $filme;
    }

    //Método Específico
    
    public function apresentarVisualizacao(){
        echo "<p> ========= Visualização ========== </p>";
        echo "Espectador: {$this->getEspectador()->getNome()}</br>";
        echo "Assistindo: {$this->getFilme()->getTitulo()} </br>";
    }
    
    //Métodos de Sobrecarga
    
    public function avaliar(){
       $this->filme->setAvaliacao(5);
    }
    
    public function avaliarNota($nota){
        $this->filme->setAvaliacao($nota);
    }
    
    public function avaliarPorcentagem($porcentagem){
        $nota = 0;
        
        if($porcentagem <= 20){
            $nota = 3;
        }elseif($porcentagem <= 50){
            $nota = 5;
        }elseif($porcentagem <= 90){
            $nota = 8;
        }else{
            $nota = 10;
        }
         
        $this->filme->setAvaliacao($nota);
       
    }

}

File: 5 - JAVA e PHP (POO)/JAVA/Aula15_Exercicio_Parte_02_Youtube/Youtube_Parte_02/src/youtube_parte_02/Visualizacao.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package youtube_parte_02;

/**
 *
 * @author ronan
 */
public class Visualizacao {
    //Atributos
    private Gafanhoto espectador;
    private Video filme;
    
    //Construtor
    public Visualizacao(Gafanhoto espectador,Video filme){
        this.setEspectador(espectador);
        this.setFilme(filme);
        this.getEspectador().viuMaisUm();
        this.getFilme().setViews(this.getFilme().getViews() + 1);
    }
    
    //Getters and Setters

    public Gafanhoto getEspectador() {
        return espectador;
    }

    public void setEspectador(Gafanhoto espectador) {
        this.espectador = espectador;
    }

    public Video getFilme() {
        return filme;
    }

    public void setFilme(Video filme) {
        this.filme = filme;
    }
    
    //Métodos Específicos
    
    public void apresentarVisualizacao(){
        System.out.println("=============== Visualização ================");
        System.out.println("Espectador: "+this.getEspectador().getNome());
        System.out.println("Asssistindo agora: "+this.getFilme().getTitulo()+"\n");
    }
    
    //Métodos de Sobrecarga
    public void avaliar(){
        this.getFilme().setAvaliacao(5);
    }
    
    public void avaliar(int nota){
        this.getFilme().setAvaliacao(nota);
    }

    public void avaliar(float porcentagem){
        if(porcentagem <= 20){
            this.getFilme().setAvaliacao(3);
        }else if(porcentagem <= 50){
            this.getFilme().setAvaliacao(5);
        }else if(porcentagem <= 90){
            this.getFilme().setAvaliacao(8);
        }else{
            this.getFilme().setAvaliacao(10);
        }
    }
    
}
